src/TimeModeling/Types/GeometryStructure.jl

```diff
@@ -21,7 +21,7 @@ mutable struct GeometryIC{T} <: Geometry
 end
 
 # Out-of-core geometry structure, contains look-up table instead of coordinates
-mutable struct GeometryOOC{T} <: Geometry
+mutable struct GeometryOOC{T} <: Geometry{T}
```

Comment:
FYI this is not how you use parametric types. T isn't used anywhere you are just creating a weird type with undefined eltype.

Reply:
This might be beneficial for future developments, i.e. developers can do

```julia

function f(g::Geometry{T})

```

no?

Reply:
It has zero use unless it actually implement a parametric type. It's is actually more dangerous/bad that good since you allow awful stuff like

```
struct test{T}
 var::Float32
end

a = test{Float64}(1f0)
```

Parametric type are not for decoration it needs to be done properly

Reply:
But in this case

```julia

abstract type Geometry{T} end

# In-core geometry structure for seismic header information
mutable struct GeometryIC{T} <: Geometry{T}
    xloc::CoordT  # Array of receiver positions (fixed for all experiments)
    yloc::CoordT
    zloc::CoordT
    dt::Array{T,1}
    nt::Array{Integer,1}
    t::Array{T,1}
end

# Out-of-core geometry structure, contains look-up table instead of coordinates
mutable struct GeometryOOC{T} <: Geometry{T}
    container::Array{SegyIO.SeisCon,1}
    dt::Array{T,1}
    nt::Array{Integer,1}
    t::Array{T,1}
    nsamples::Array{Integer,1}
    key::String
    segy_depth_key::String
end

```

both of its subtypes (GeometryIC and GeometryOOC) are typed by T. Then this

```julia
function f(g::Geometry{T})
```
actually makes sense because it forces `g` to either come from `GeometryIC{T}` or `GeometryOOC{T}`, right? If `g` is `GeometryIC{T1}` and `T1 != T` then it doesn't fall into this function `f`